### PR TITLE
Switch to rspec-puppet-facts

### PR DESCRIPTION
**Pull request by ekohl**

### CODE REVIEW

File: spec/spec_helper_local.rb [hunk 1]
```diff
@@ -1,82 +1,6 @@
...
-  end
-end
-
-def with_distro_facts(distro)
-  send("with_#{distro.downcase}_facts")
-end
+require 'rspec-puppet-facts'
+include RspecPuppetFacts
+                                                    # Original fact sources:
+add_custom_fact :puppetversion, Puppet.version      # Facter, but excluded from rspec-puppet-facts
```
Comment (wyardley) [inline spec/spec_helper_local.rb]: Is there a more up to date way to include these (maybe in `spec/default_module_facts.yaml`)?

File: spec/classes/rabbitmq_spec.rb [hunk 2]
```diff
@@ -9,312 +9,187 @@
...
-  end
+  on_supported_os.each do |os, facts|
+    context "on #{os}" do
+      let(:facts) { facts }
 
-  context 'on Debian' do
-    let(:params) { { repos_ensure: true } }
+      has_systemd = (
+        (facts[:osfamily] == 'RedHat' && facts[:os]["release"]["major"].to_i >= 7) ||
+        (facts[:osfamily] == 'Debian' && facts[:os]["release"]["full"] == '16.04')
```
Comment (bastelfreak) [inline spec/classes/rabbitmq_spec.rb]: can you add archlinux here as well?

File: spec/spec_helper_local.rb (hunk 1 above)
Comment (bastelfreak) [reply spec/spec_helper_local.rb]: after a modulesync, puppetversion will be set. staging:http_get can be set via spec/default_module_facts.yml

File: spec/classes/rabbitmq_spec.rb (hunk 2 above)
Comment (bastelfreak) [reply spec/classes/rabbitmq_spec.rb]: I will add arch suport in a separate PR

**Comment by wyardley** [comment]
Did introduce some rubocop warnings, which I'll fix  while rebasing #580

**Pull request merged**